### PR TITLE
Summary UI android Agent App

### PR DESCRIPTION
**Pull request by prateek-kommunicate**
**Logic**
Storing the message if the app is Agent App and Message has metadata of KM_SMMARY to true. If either of these condition became false then the SUMMARY UI will not be shown to the user. 

**UI**
<video src="https://github.com/user-attachments/assets/64f463c8-fa09-4b8c-b56b-27db0459cfee" />

### CODE REVIEW

File: kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/MobiComConversationService.java [hunk 1]
```diff
@@ -442,8 +444,23 @@ public synchronized List<Message> getMessages(Long startTime, Long endTime, Cont
...
-                        continue;
+
+
+                    if (Message.MetaDataType.HIDDEN.getValue().equals(message.getMetaDataValueForKey(Message.MetaDataType.KEY.getValue()))
+                            || Message.MetaDataType.PUSHNOTIFICATION.getValue().equals(message.getMetaDataValueForKey(Message.MetaDataType.KEY.getValue()))
+                    ) {
+                        // Check if message is conversation summary.
+                        boolean isKmSummary;
+                        if (message.getMetadata().containsKey(KM_SUMMARY)) {
+                             isKmSummary = Boolean.parseBoolean(message.getMetaDataValueForKey(KM_SUMMARY));
```
Comment (adarshmishra) [inline kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/MobiComConversationService.java:455]: Optimize code, try something like: 
  boolean isKmSummary = message.getMetadata().containsKey(KM_SUMMARY) && Boolean.parseBoolean(message.getMetaDataValueForKey(KM_SUMMARY));

File: kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/MobiComConversationService.java [hunk 2]
```diff
@@ -345,7 +347,7 @@ public synchronized List<Message> getMessagesForParticularThread(Long startTime,
     }
 
 
-    public synchronized List<Message> getMessages(Long startTime, Long endTime, Contact contact, Channel channel, Integer conversationId, boolean isSkipRead, boolean isForSearch) {
+    public synchronized List<Message> getMessages(Boolean isAgentApp, Long startTime, Long endTime, Contact contact, Channel channel, Integer conversationId, boolean isSkipRead, boolean isForSearch) {
```
Comment (adarshmishra) [inline kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/MobiComConversationService.java:350]: Check if we need to pass it as an argument or find some better way to identify if it is an agent app. Or changes it to something generic like, show summaryView or customView

Comment (prateek-kommunicate) [reply kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/MobiComConversationService.java]: Done, Added in [2410150](https://github.com/Kommunicate-io/Kommunicate-Android-Chat-SDK/pull/565/commits/2410150a3bea2d2b2d2dfd0479b5012cc515f3d8)

File: kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/MobiComConversationService.java (hunk 1 above)
Comment (prateek-kommunicate) [reply kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/MobiComConversationService.java]: Done Added in [2410150](https://github.com/Kommunicate-io/Kommunicate-Android-Chat-SDK/pull/565/commits/2410150a3bea2d2b2d2dfd0479b5012cc515f3d8)

**Review by adarshmishra — APPROVED** [review]

**Pull request merged**